### DISCUSSION TITLE
Login panel?

### DISCUSSION THREAD

**Discussion by pawisoon** [PrivateBin/PrivateBin#1106] [Ideas]
Hi there, 
Just wondering is there any way to close the privatebin behing login page? 
I would like only registered users to use privatebin, instead being accesible by anons. 

Any tips if it's already implemented? 
Cheers!

**Comment by elrido** [comment]
PrivateBin itself is fundamentally designed to operate without any concept of users and user permissions (see the [Threat Model](https://github.com/PrivateBin/PrivateBin/wiki/Threat-Model) for an in-depth analysis), but nothing stops you from using a third-party solution to add that layer of functionality, just like with any other non-authenticating web application.

You'd start by [restricting the upload, like outline here for nginx](https://github.com/PrivateBin/PrivateBin/wiki/Restrict-upload-using-NGINX), and instead of restricting it by IP and/or requiring credentials, you'd redirect the requests to your authentication solution of choice, which handles the login in any way they chose. But how you'd setup such a solutions and which one to pick is really out of scope for PrivateBin, IMHO.
